tests/src/test/java/me/snowdrop/istio/dt/OpenshiftIT.java

```diff
@@ -0,0 +1,134 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+
+@RunWith(Arquillian.class)
+public class OpenshiftIT {
+    @ArquillianResource
+    private IstioAssistant istioAssistant;
+
+    @ArquillianResource
+    private OpenShiftAssistant openShiftAssistant;
+
+    private static String istioURL;
+    private static String jaegerURL;
+    private static boolean isInit = false;
+    private static List<IstioResource> routeRule = null;
+
+    private final String appUrl = "greeting/";
+
+    @Before
+    public void init() throws Exception {
+        // initiate only before first test
+        if (!isInit) {
+            istioURL = "http://" + getIstioRoute("istio-ingress");
+            jaegerURL = "https://" + getIstioRoute("jaeger-query") + "api/traces";
+
+            routeRule = deployRouteRule("route-rule-redir.yml");
+            await().
+                    pollInterval(1, TimeUnit.SECONDS)
+                    .atMost(1, TimeUnit.MINUTES)
+                    .until(() -> {
+                try {
+                    Response response = RestAssured.get(istioURL + appUrl);
+                    return response.getStatusCode() == 200;
+                } catch (Exception ignored) {
+                    return false;
+                }
+            });
+
+            // jaeger is using HTTPS with self-signed certificate
+            // allow acceptance of non-trusty certificates
+            RestAssured.useRelaxedHTTPSValidation();
+
+            isInit = true;
+        }
+    }
+
+    @After
+    public void cleanup() {
+        if (routeRule != null){
+            istioAssistant.undeployIstioResources(routeRule);
+        }
+    }
+
+    /**
+     * Interacts with application and test if traces are present in jaeger
+     * @throws ParseException
+     * @throws InterruptedException
+     */
+    @Test
+    public void tracingTest() throws ParseException, InterruptedException {
+        Long startTime = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis());
+
+        RestAssured
+                .get(istioURL + "greeting/api/greeting");
+
+        TimeUnit.SECONDS.sleep(10); // wait 10 seconds so span will show up in the jaeger
+
+        JSONArray jaegerTraces = getJaegerTraces(startTime);
+        Assert.assertEquals("Query should create one trace",1,jaegerTraces.size());
+        int spanCount = ((JSONArray)((JSONObject)jaegerTraces.get(0)).get("spans")).size();
+        Assert.assertTrue("Trace should have at least 4 spans, has: " + spanCount, spanCount >= 4);
+    }
+
+    private List<IstioResource> deployRouteRule(String routeRuleFile) throws IOException {
+        return istioAssistant.deployIstioResources(
+                Files.newInputStream(Paths.get("../rules/" + routeRuleFile)));
+    }
+
+    /**
+     * Get a route from istio namespace
+     * @param routeName name of the route to get
```

Comment:
Please capitalize `route`, use `Route` to make it clear that we are talking about the Openshift Route.